contracts/src/token/memecoin.cairo

```diff
@@ -61,12 +81,16 @@ mod UnruggableMemecoin {
     struct Storage {
         marker_v_0: (),
         team_allocation: u256,
+        pre_launch_holders_count: u8,
         tx_hash_tracker: LegacyMap<ContractAddress, felt252>,
         transfer_restriction_delay: u64,
         launch_time: u64,
+        launch_block_number: u64,
```

Comment:
what for?

Reply:
So I can get the USDC price of the quote amount at that block number and compute the starting mcap

Reply:
isn't there a way to do that with the launch time only?